src/components/singlEventPageComponents/EventDetails.jsx

```diff
@@ -1,7 +1,150 @@
-import React from "react";
+import { doc, onSnapshot } from "firebase/firestore";
+import Image from "next/image";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import React, { useEffect, useState } from "react";
+
+import { firestore } from "@/lib/firebase/controller";
+
+import Loader from "../loader/Loader";
 
 function EventDetails() {
-    return <div>extract details from eventForm in reusableComponents</div>;
+    const [eventDisplay, setEventDisplay] = useState({});
+
+    const router = useRouter();
+    const { id } = router.query;
+
+    useEffect(() => {
+        if (id) {
+            const eventRef = doc(firestore, "events", id);
+            onSnapshot(eventRef, (docEvent) => {
```

Comment:
walk me throught it

Reply:
it's conditional rendering where the doc ref (the event) wont be fetched and displayed to the user untill the id fully exists.
this way the when we refresh and incase there it's taking too long to get the id, errors wont be displayed (like in the previous) instead wait for the id and then showcase.
i hope this made sense